pepper-apis/src/main/java/org/broadinstitute/ddp/service/ActivityInstanceService.java

```diff
@@ -77,4 +112,311 @@ public ActivityInstanceService(ActivityInstanceDao actInstanceDao, PexInterprete
         return getTranslatedActivity(handle, userGuid, operatorGuid, ActivityType.FORMS,
                 formInstanceGuid, isoLangCode, style).map(typeChecker);
     }
+
+    /**
+     * Find list of activity instance summaries for user and study. List will be sorted by activity display order
+     * (ascending) and instance creation time (descending).
+     *
+     * <p>The instance summaries will be translated to the user's preferred language, or falls back to study's default
+     * language if preferred language is not available. If study doesn't have a default language, English will be used
+     * as the fallback.
+     *
+     * <p>Some computed properties such as instance numbering, question/answer count, or rendered strings will not be
+     * set. Caller is responsible for performing the computations and setting them properly.
+     *
+     * @param handle            database handle
+     * @param userGuid          GUID of the user to get activity instance summaries for
+     * @param studyGuid         GUID of the study to get activity instance summaries for
+     * @param preferredLangCode The desired translation language
+     * @return Sorted list of activity instance summaries for the user/study
+     */
+    public List<ActivityInstanceSummary> listTranslatedInstanceSummaries(Handle handle,
+                                                                         String userGuid,
+                                                                         String studyGuid,
+                                                                         String preferredLangCode) {
+        List<ActivityInstanceSummaryDto> summaryDtos = handle
+                .attach(org.broadinstitute.ddp.db.dao.ActivityInstanceDao.class)
```

Comment:
Not most important thing ever, but is package name needed here?

Reply:
Yes... because we haven't completely refactored our code so we have two `ActivityInstanceDao` classes... one in `ddp.db` package and another in `ddp.db.dao` package. Both are used here, so need to qualify the import.